tests/test_cargo_init.rs

```
@@ -0,0 +1,297 @@
+#![allow(unused_imports)]
+use std::fs::{self, File};
+use std::io::prelude::*;
+use std::env;
+use tempdir::TempDir;
+
+use support::{execs, paths, cargo_dir};
+use support::paths::CargoPathExt;
+use hamcrest::{assert_that, existing_file, existing_dir, is_not};
+
+use cargo::util::{process, ProcessBuilder};
+
+fn setup() {
+}
+
+fn cargo_process(s: &str) -> ProcessBuilder {
+    let mut p = process(&cargo_dir().join("cargo"));
+    p.arg(s).cwd(&paths::root()).env("HOME", &paths::home());
+    return p;
+}
+
+test!(simple_lib {
+    assert_that(cargo_process("init").arg("--vcs").arg("none")
+                                    .env("USER", "foo"),
+                execs().with_status(0));
+
+    assert_that(&paths::root().join("Cargo.toml"), existing_file());
+    assert_that(&paths::root().join("src/lib.rs"), existing_file());
+    assert_that(&paths::root().join(".gitignore"), is_not(existing_file()));
+
+    assert_that(cargo_process("build").cwd(&paths::root().join(".")),
+                execs().with_status(0));
+});
+
+test!(simple_bin {
+    let path = paths::root().join("foo");
+    fs::create_dir(&path).ok();
```

This should be `.unwrap()` to not hide a failure